draft-js-inline-toolbar-plugin/src/components/Toolbar/index.js

```diff
@@ -45,25 +63,103 @@ export default class Toolbar extends React.Component {
    */
   onOverrideContent = (overrideContent) => {
     this.setState({ overrideContent });
-  }
+  };
 
   onSelectionChanged = () => {
     // need to wait a tick for window.getSelection() to be accurate
     // when focusing editor with already present selection
     setTimeout(() => {
       if (!this.toolbar) return;
-      const relativeParent = getRelativeParent(this.toolbar.parentElement);
-      const toolbarHeight = this.toolbar.clientHeight;
-      const relativeRect = (relativeParent || document.body).getBoundingClientRect();
+
+      if (typeof this.state.width !== 'number') {
+        this.setState({ width: this.toolbar.offsetWidth });
+      } else {
+        // toolbar width must forcibly overwritten to prevent unexpected geometry
+        // changes
+        this.toolbar.style.width = `${this.state.width}px`;
+      }
+
+      const metrics = {
+        rule: 'left',
+        ruleValue: 0,
+        shift: null,
+        shiftValue: 0
+      };
+      // boundings of the selected text
       const selectionRect = getVisibleSelectionRect(window);
+      const selection = this.props.store.getItem('getEditorState')().getSelection();
+      if (!selectionRect || selection.isCollapsed()) return;
 
-      if (!selectionRect) return;
+      const relativeParent = getRelativeParent(this.toolbar.parentElement);
+      const relativeRect = (relativeParent || document.body).getBoundingClientRect();
+      const windowWidth = document.documentElement.clientWidth;
+      // we should take into account a case when we don't have relative parent,
+      // but our body has a margin
+      const bodyMargin = relativeParent ? 0 : getMargin(document.body);
```

Comment:
what I meant here was that we should leave this completely in user land.  I don't think we should support this usecase.

Reply:
Hi, this is not related with previous conversation. This is new bugfix. It is related with cases like we have in storybook, when body has a margin, and parent not relative. We should take into account this body margin in that case.